src/qibolab/instruments/dummy.py

```diff
@@ -1,15 +1,85 @@
+from dataclasses import dataclass, field
 from typing import Dict, List, Union
 
 import numpy as np
 from qibo.config import log
 
 from qibolab import AcquisitionType, AveragingMode, ExecutionParameters
 from qibolab.instruments.abstract import Controller
+from qibolab.instruments.port import Port
 from qibolab.platform import Qubit
 from qibolab.pulses import PulseSequence
 from qibolab.sweeper import Sweeper
 
 
+@dataclass
+class DummyPort(Port):
+    name: str
+    _offset: float = 0.0
+    _lo_frequency: int = 0
+    _lo_power: int = 0
+    _gain: int = 0
+    _attenuation: int = 0
+    _power_range: int = 0
+    _filter: dict = field(default_factory=dict)
+
+    @property
+    def offset(self):
+        return self._offset
+
+    @offset.setter
+    def offset(self, value):
+        self._offset = value
```

Comment:
I wonder if just defining `self.offset` should be enough. It won't be a property any longer, but a property emulates an attribute anyhow, so maybe it's just a simpler alternative.

Reply:
Indeed, you may have noticed that this is the case for most properties, in most `Port` implementations. The reason I am not "downgrading" to attributes and keep them as properties, is that there may be a driver (most likely qblox) where the setter won't be trivial and you will need to define it explicitly as a function. And I want the interface to be the same for all drivers.

If all the setters are trivial, maybe we don't need `Port` objects at all. Note that we introduced them to support qblox and enable removing the `MultiqubitPlatform`. The other drivers are so far working without them without any issues.

Reply:
I'm not suggesting to downgrade everywhere, the idea was that properties and attributes may be compatible: they have the exact same interface, so it should be fine to have a property `.myprop` in the parent class, but using a `.myprop` attribute in a child.

If there is no reason they have to be properties in the parent, then you could consider even making attributes there, and only use properties when strictly needed.
But raising errors in the parent is not a terrible idea, even though I see a lot of duplication (maybe we could compress it a bit, but the idea should be fine).

Reply:
Thanks, I converted all these properties to attributes. I had the impression that overriding a property with an attribute is not safe, but it seems to work.

Reply:
I still have the same *impression*, like there was something wrong. But in principle if it's working there should be nothing bad. And it seems like, even though it still calls the property for getting and setting.

So, maybe it's better to have plain attributes in the base class as well, since overwriting them with properties should work, but overwriting the property with the attribute is *working*, but still using the property itself. So, if the original property was raising, it should still raise.